src/main/java/org/jabref/logic/importer/fetcher/CitationRelationFetcher.java

```diff
@@ -0,0 +1,137 @@
+     */
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) throws FetcherException {
+        String doi = entry.getField(StandardField.DOI).orElse("");
+        if (searchType != null) {
+            List<BibEntry> list = new ArrayList<>();
+            try {
+                LOGGER.info("Search: " + BASIC_URL + doi);
+                JSONArray json = readJsonFromUrl(BASIC_URL + doi);
+                if (json == null) {
+                    throw new FetcherException("No internet connection! Please try again.");
+                } else if (json.isEmpty()) {
+                    return list;
+                }
+                LOGGER.info("API Answer: " + json.toString());
+                String[] items = json.getJSONObject(0).getString(searchType.label).split("; ");
+                if (!Arrays.equals(items, new String[]{""})) {
+                    int i = 1;
+                    for (String item : items) {
+                        LOGGER.info("Current Item " + i + "/" + items.length);
+                        setProgress(i, items.length);
+                        if (!doi.equals(item) && !item.equals("")) {
+                            DoiFetcher doiFetcher = new DoiFetcher(JabRefPreferences.getInstance().getImportFormatPreferences());
+                            try {
+                                doiFetcher.performSearchById(item).ifPresent(list::add);
+                            } catch (FetcherException fetcherException) {
+                                // No information for doi found
+                            }
+                        }
+                        i++;
+                    }
+                }
+            } catch (IOException | JSONException e) {
+                throw new FetcherException("Couldn't connect to opencitations.net! Please try again.");
+            }
+            return list;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Method to read JSON files from URL
+     *
+     * @param url API URL to search
+     * @return JSONArray containing the response of the API
+     */
+    public static JSONArray readJsonFromUrl(String url) throws IOException, JSONException {
+        try (InputStream is = new URL(url).openStream()) {
+            BufferedReader rd = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
+            String jsonText = readAll(rd);
+            return new JSONArray(jsonText);
+        } catch (UnknownHostException | SocketException exception) {
+            return null;
+        }
+    }
+
+    private static String readAll(Reader rd) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        int cp;
+        while ((cp = rd.read()) != -1) {
+            sb.append((char) cp);
+        }
+        return sb.toString();
+    }
+
+    private void setProgress(double i, double total) {
+        progress.set(i/total);
+    }
+
+    public DoubleProperty getProgress() {
```

Comment:
I'm not sure if the fetcher should report it's progress. How long does it usually take to fetch the citation information?

Reply:
The opencitationsapi usualloy answers quite fast, the bottleneck is the DOI-Fetcher we use to get the Bibliographic Information for each DOI... The time the BackgroundTask needs to fetch all of the Information, heavily depends on the numer of DOI's... It also seems like the performance of the fetcher changes from time to time...
I think the Progressindication makes sense anyway, because the users gets informed, that something happens. The User can be sure to see all the related articles after the serachprocess finished